src/Features/Core/Portable/ExtractMethod/MethodExtractor.CodeGenerator.cs

```diff
@@ -80,16 +84,36 @@ public async Task<GeneratedCode> GenerateAsync(CancellationToken cancellationTok
                 var newCallSiteRoot = callSiteDocument.Root;
                 var previousMemberNode = GetPreviousMember(callSiteDocument);
 
-                // it is possible in a script file case where there is no previous member. in that case, insert new text into top level script
-                var destination = (previousMemberNode.Parent == null) ? previousMemberNode : previousMemberNode.Parent;
+                SyntaxNode destination;
+                if (ExtractLocalFunction)
+                {
+                    destination = InsertionPoint.With(callSiteDocument).GetContext();
+                }
+                else
+                {
+                    // it is possible in a script file case where there is no previous member. in that case, insert new text into top level script
+                    destination = previousMemberNode.Parent ?? previousMemberNode;
+                }
 
                 var codeGenerationService = SemanticDocument.Document.GetLanguageService<ICodeGenerationService>();
 
                 var result = GenerateMethodDefinition(cancellationToken);
-                var newContainer = codeGenerationService.AddMethod(
-                    destination, result.Data,
-                    new CodeGenerationOptions(afterThisLocation: previousMemberNode.GetLocation(), generateDefaultAccessibility: true, generateMethodBodies: true),
-                    cancellationToken);
+
+                SyntaxNode newContainer;
```

Comment:
Can we group together the different variables that depend on `ExtractLocalFunction`? `newContainer` and `destination` both have if/else blocks for it